tests/Unit/Execution/Arguments/ArgumentSetTest.php

```diff
@@ -120,4 +126,64 @@ public function testListOfInputObjectsToArray(): void
             $argumentSet->toArray()
         );
     }
+
+    public function testRenameInput(): void
+    {
+        $generateRenameDirective = function ($newName) {
+            $node = new DirectiveDefinitionNode([]);
+
+            $node->directives = [
+                new DirectiveNode([
+                    'name' => new NameNode(['value' => 'rename']),
+                    'arguments' => [
+                        new ArgumentNode([
+                            'name' => new NameNode(['value' => 'attribute']),
+                            'value' => new StringValueNode(['value' => $newName]),
+                        ]),
+                    ],
+                ]),
+            ];
+
+            $rename = new RenameDirective();
+
+            return $rename->hydrate($node);
+        };
+
+        $firstName = new Argument();
+        $firstName->value = 'Michael';
+        $firstName->directives = collect([$generateRenameDirective('first_name')]);
+
+        $lastName = new Argument();
+        $lastName->value = 'Jordan';
+        $lastName->directives = collect([$generateRenameDirective('last_name')]);
+
+        $postName = new Argument();
+        $postName->value = 'Hello World';
+        $postName->directives = collect([$generateRenameDirective('title')]);
+
+        $postSet = new ArgumentSet();
+        $postSet->arguments['post_title'] = $postName;
+
+        $postArg = new Argument();
+        $postArg->value = $postSet;
+
+        $userSet = new ArgumentSet();
+        $userSet->arguments = [
+            'firstName' => $firstName,
+            'lastName' => $lastName,
+            'post' => $postArg,
+        ];
+
+        $renamedSet = $userSet->rename();
+
+        $this->assertSame([
+            'first_name' => $firstName,
```

Comment:
What are the three different args testing?

The tests should cover more different cases rather then repeating the same test case.